src/isomorphic/classic/element/ReactElementValidator.js

```diff
@@ -97,10 +97,18 @@ function validateExplicitKey(element, parentType) {
   }
   element._store.validated = true;
 
-  warnAndMonitorForKeyUse(
-    'Each child in an array or iterator should have a unique "key" prop.',
-    element,
-    parentType
+  var addendums = getAddendumsForKeyUse('uniqueKey', element, parentType);
+  if (addendums === null) {
+    // we already showed the warning
+    return;
+  }
+  warning(
+    false,
+    'Each child in an array or iterator should have a unique "key" prop.' +
+    '%s%s%s',
+    addendums.parentOrOwner || '',
```

Comment:
Wonder if it's better to merge them into one string first. I guess this is better for analyzing later.